Spigot-Server-Patches/0522-Add-permission-for-command-blocks.patch

```diff
@@ -5,58 +5,58 @@ Subject: [PATCH] Add permission for command blocks
 
 
 diff --git a/src/main/java/net/minecraft/server/BlockCommand.java b/src/main/java/net/minecraft/server/BlockCommand.java
-index d0f75c8f93d3ceb1ee358baeb51f64e7e575bbc0..48cc48bd999c02268fa3270401c0df77a327ab13 100644
+index dd7066d1a72f5c6f54c1f40a7b694deb827410dc..6b353a99c04e0312f520f8559c05ddaf51c26aaf 100644
 --- a/src/main/java/net/minecraft/server/BlockCommand.java
 +++ b/src/main/java/net/minecraft/server/BlockCommand.java
-@@ -110,7 +110,7 @@ public class BlockCommand extends BlockTileEntity {
+@@ -105,7 +105,7 @@ public class BlockCommand extends BlockTileEntity {
      public EnumInteractionResult interact(IBlockData iblockdata, World world, BlockPosition blockposition, EntityHuman entityhuman, EnumHand enumhand, MovingObjectPositionBlock movingobjectpositionblock) {
          TileEntity tileentity = world.getTileEntity(blockposition);
  
 -        if (tileentity instanceof TileEntityCommand && entityhuman.isCreativeAndOp()) {
 +        if (tileentity instanceof TileEntityCommand && (entityhuman.isCreativeAndOp() || (entityhuman.isCreative() && entityhuman.getBukkitEntity().hasPermission("minecraft.commandblock")))) { // Paper - command block permission
              entityhuman.a((TileEntityCommand) tileentity);
-             return EnumInteractionResult.SUCCESS;
+             return EnumInteractionResult.a(world.isClientSide);
```

Comment:
This seems like something that would benefit from an obf helper.

Reply:
we don't touch those lines, its irrelevent